sites/game-client/ui/setting-modal.mjs

```diff
@@ -0,0 +1,291 @@
+// Copyright 2021 HITCON Online Contributors
+// SPDX-License-Identifier: BSD-2-Clause
+
+import Modal from './modal.mjs';
+
+const SETTING_DIV = 'setting-overlay';
+
+/**
+ * The setting modal.
+ */
+class SettingModal extends Modal {
+  /**
+   * Create a setting page.
+   * @param {MainUI} mainUI
+   * @constructor
+   */
+  constructor(mainUI) {
+    const DOM = document.getElementById(SETTING_DIV);
+    super(mainUI, DOM);
+    this.mainUI = mainUI;
+    this.sections = {};
+    this.mainDOM = document.getElementById('setting-main');
+
+    this.initSetting();
+
+    document.getElementById('setting-reset').addEventListener('click', () => {
+      this.reset();
+    });
+
+    document.getElementById('setting-save').addEventListener('click', () => {
+      this.save();
+    });
+  }
+
+  /**
+   * Create basic setting items.
+   */
+  initSetting() {
+    this.addSection('general', '一般設定', 0);
+    this.addSubsection('general', 'account', '帳號設定', 0);
+    this.addTextInput('account', 'nickname', '暱稱', 'Eevee', 0);
+
+    this.addSubsection('general', 'device', '裝置設定', 10);
+    this.addDropdown('device', 'video', '影像輸入', 0);
+    this.updateDropdownOptions('device', 'video', {
+      'facetime': 'FaceTime',
+      'obs': 'OBS Virtual Camera',
+    }, 'obs');
+    this.addDropdown('device', 'audio', '音訊輸入', 10);
+    this.updateDropdownOptions('device', 'audio', {
+      'builtin': 'Built-in Microphone',
+      'headphone': 'External Headphone',
+      'bluetooth': 'Bluetooth device',
+    }, 'builtin');
+
+    this.addSwitch('general', 'data_collect', '去識別化資料蒐集', true, 50);
+
+    this.addSection('chat', '聊天設定', 10);
+    this.addDropdown('chat', 'example', '範例下拉選單', 0);
+    this.updateDropdownOptions('chat', 'example', {
+      'default': 'Default',
+      'meow': 'Meow',
+    }, 'default')
+  }
+
+  /**
+   * @param {string} name
+   * @param {string} displayName
+   * @param {number} order
+   */
+  addSection(name, displayName, order) {
```

Comment:
Optional: Ideally a `addSection` could return an object that the caller can act on and add other items. Same with `addSubsection`.